src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  FC,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo, UpdateTodoArgs } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  removeTodo: (todoId: number) => void;
+  editTodo: (todoId: number, args: UpdateTodoArgs) => Promise<void>;
+  loadingItems: number[];
+  isLoadingAddTodo: boolean;
+}
+
+export const TodoItem: FC<Props> = ({
+  todo,
+  removeTodo,
+  editTodo,
+  loadingItems,
+  isLoadingAddTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const isLoading = loadingItems.includes(todo.id)
+    || (isLoadingAddTodo && todo.id === 0);
+
+  useEffect(() => {
+    if (!inputRef.current && !isEditing) {
+      return;
+    }
+
+    inputRef.current?.focus();
+  }, [isEditing, isLoading]);
+
+  const handleCompleteChange = async () => {
+    await editTodo(
+      todo.id,
+      { completed: !todo.completed },
+    );
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleTitleChange = async () => {
+    if (newTitle === todo.title) {
+      setIsEditing(false);
+
+      return;
+    }
+
+    if (!newTitle) {
+      removeTodo(todo.id);
+
+      return;
+    }
+
+    await editTodo(
+      todo.id,
+      { title: newTitle },
+    );
+
+    setIsEditing(false);
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.code !== 'Escape') {
+      return;
+    }
```

Comment:
When I click `ESC` edit should be canceled and return to the previous value
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/a9881fb7-3210-4c13-ad0a-0afd3a4e761f)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/f09c3cae-3983-47ff-b6b8-f6ff91e2d870)

Reply:
@33Orange yeap, it's my mistake, I thought that changes should be saved